pandas/core/construction.py

```diff
@@ -412,6 +419,11 @@ def extract_array(obj: object, extract_numpy: bool = False) -> Union[Any, ArrayL
     array([1, 2, 3])
     """
     if isinstance(obj, (ABCIndex, ABCSeries)):
+        if isinstance(obj, ABCRangeIndex):
+            if extract_range:
```

Comment:
why would we not *always* want to do this

Reply:
does it not make sense to always do this and simply wrap a `np.asarray` there?

Reply:
if it were just one place then maybe, but i also think this could be relevant in Series._arith_method et al